pkg/handlers/v1/hydrator.go

```diff
@@ -0,0 +1,78 @@
+package v1
+
+import (
+	"context"
+	"time"
+
+	"github.com/asecurityteam/nexpose-vuln-hydrator/pkg/domain"
+)
+
+// HydrationHandler consumes AssetEvents, hydrates them with vulnerability details,
+// and returns an AssetVulnerabilitiesEvent or error
+type HydrationHandler struct {
+	Hydrator domain.Hydrator
+}
+
+// AssetEvent contains JSON annotations for scanned Asset events
+type AssetEvent struct {
+	LastScanned time.Time `json:"lastScanned"`
+	ID          int64     `json:"id"`
+	IP          string    `json:"ip"`
+	Hostname    string    `json:"hostname"`
+}
+
+// AssetVulnerabilitiesEvent contains the Asset event hydrated with vulnerability details
+type AssetVulnerabilitiesEvent struct {
+	AssetEvent
+	AssetVulnerabilityDetails []AssetVulnerabilityDetails `json:"assetVulnerabilityDetails"`
+}
+
+// AssetVulnerabilityDetails contains pertinent information about a vulnerability
+type AssetVulnerabilityDetails struct {
+	ID             string             `json:"id"`
+	Results        []AssessmentResult `json:"results"`
+	CvssV2Score    float64            `json:"cvssV2Score"`
+	CvssV2Severity string             `json:"cvssV2Severity"`
+	Description    string             `json:"description"`
+	Title          string             `json:"title"`
+	Solutions      []string           `json:"solutions"`
+}
+
+// AssessmentResult contains information about the port/protocol the vulnerability was discovered on
+type AssessmentResult struct {
+	Port     int32  `json:"port"`
+	Protocol string `json:"protocol"`
+}
+
+// Handle accepts AssetEvents and returns a hydrated asset containing vulnerability information
+func (h *HydrationHandler) Handle(ctx context.Context, evt AssetEvent) (AssetVulnerabilitiesEvent, error) {
+	asset := domain.Asset(evt)
+	assetWithVulnerabilityDetails, err := h.Hydrator.HydrateVulnerabilities(asset)
+	if err != nil {
+		return AssetVulnerabilitiesEvent{}, err
```

Comment:
question: would an error log be suitable here?

Reply:
I can add that - a good idea to get some of the mocking boilerplate in for future PRs